x/txfees/keeper/msg_server.go

```diff
@@ -0,0 +1,31 @@
+package keeper
+
+import (
+	"context"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+
+	"github.com/osmosis-labs/osmosis/v24/x/txfees/types"
+)
+
+type msgServer struct {
+	keeper *Keeper
+}
+
+func NewMsgServerImpl(keeper *Keeper) types.MsgServer {
+	return &msgServer{
+		keeper: keeper,
+	}
+}
+
+// SetFeeTokens sets the provided fee tokens for the chain. The sender must be whitelisted to set fee tokens.
+func (server msgServer) SetFeeTokens(goCtx context.Context, msg *types.MsgSetFeeTokens) (*types.MsgSetFeeTokensResponse, error) {
```

Comment:
Would it make sense to have `AddFeeTokens` and `RemoveFeeTokens`

I think that the current UX of having to repeat all fee tokens is tedious and risky because it is easy to miss a previously existing token

Reply:
@p0mvn  This acts as both a add and remove fee tokens, this should not require passing in all previously existing tokens. I will wait for you to ACK that I am not misunderstanding before I merge

Reply:
@p0mvn this method iterates over the setFeeToken call here https://github.com/osmosis-labs/osmosis/blob/00b080a665874350ad82530124b28ecb3ce88921/x/txfees/keeper/feetokens.go#L136-L158

If the token's poolId is set to 0, it is deleted. If it is new, it is added to the prefix store. Therefore this method acts as both an AddFeeToken and RemoveFeeToken, and there is no requirement to provide the entire fee token set each time. Please ACK that you agree and then I will merge.

Reply:
ACK.

I meant something else - to make sure that we don't have to specify all the tokens given the "set" abstraction. That is, that "Set" does not accidentally remove tokens not listed as part of the input.

Upon reviewing the code, this is not a problem.